src/seraphis_core/legacy_decoy_selector_flat.cpp

```diff
@@ -45,50 +45,51 @@
 namespace sp
 {
 //-------------------------------------------------------------------------------------------------------------------
-LegacyDecoySelectorFlat::LegacyDecoySelectorFlat(const std::uint64_t min_index, const std::uint64_t max_index) :
-    m_min_index{min_index},
-    m_max_index{max_index}
+LegacyDecoySelectorFlat::LegacyDecoySelectorFlat(index_bounds_by_amount_t index_bounds_by_amount) :
+    m_index_bounds_by_amount(std::move(index_bounds_by_amount))
 {
     // checks
-    CHECK_AND_ASSERT_THROW_MES(m_max_index >= m_min_index, "legacy decoy selector (flat): invalid element range.");
+    for (const auto &p : m_index_bounds_by_amount)
+    {
+        const std::uint64_t min_ind{p.second.first};
+        const std::uint64_t max_ind{p.second.second};
+        CHECK_AND_ASSERT_THROW_MES(max_ind >= min_ind, "legacy decoy selector (flat): min > max index.");
+    }
 }
 //-------------------------------------------------------------------------------------------------------------------
-void LegacyDecoySelectorFlat::get_ring_members(const std::uint64_t real_ring_member_index,
+void LegacyDecoySelectorFlat::get_ring_members(const legacy_output_index_t real_ring_member_index,
         const std::uint64_t num_ring_members,
-        std::vector<std::uint64_t> &ring_members_out,
+        std::set<legacy_output_index_t> &ring_members_out,
         std::uint64_t &real_ring_member_index_in_ref_set_out) const
 {
-    CHECK_AND_ASSERT_THROW_MES(real_ring_member_index >= m_min_index,
+    const rct::xmr_amount amount{real_ring_member_index.ledger_indexing_amount};
+    const std::uint64_t min_index{get_min_index(amount)};
+    const std::uint64_t max_index{get_max_index(amount)};
```

Comment:
```suggestion
    const std::uint64_t min_index{this->get_min_index(amount)};
    const std::uint64_t max_index{this->get_max_index(amount)};
```
Without this it is ambiguous where the function comes from/how it could do what you want.